python/dgl/distributed/dist_graph.py

```diff
@@ -336,6 +336,14 @@ def __init__(self, server_id, ip_config, num_servers,
             self.client_g, _, _, self.gpb, graph_name, \
                     ntypes, etypes = load_partition(part_config, self.part_id, load_feats=False)
             print('load ' + graph_name)
+            # formatting dtype
+            for k, dtype in FIELD_DICT.items():
+                if k in self.client_g.ndata:
+                    self.client_g.ndata[k] = F.tensor(
+                        self.client_g.ndata[k], dtype=dtype)
```

Comment:
Use `F.astype` which will save a deep copy if the dtype matches the provided one.